src/simulate.jl

```diff
@@ -151,7 +332,7 @@ vector is generated as `randn(rng, size(M, 2))`
 """
 function unscaledre! end
```

Comment:
Perhaps we should change this name for some of the methods.  I kept trying to remember why it was called `unscaled` and realized that this only applies to the case where the random effects are being simulated so they get multiplied by λ but not by σ.  But the name doesn't apply to the cases shown here where `b` is being passed in.

Reply:
I realized what was bothering me about this.  The methods such as
```julia
unscaledre!(y, A, b)
```
could be written as 5-argument `mul!` methods.  Essentially the operation is
```julia
mul!(y, A, b, one(T), one(T))
```

I plan to make a modification where the `unscaledre!` methods are only those that simulate the data and the 5-argument `mul!` methods are in `src/remat.jl`.  Should I create a new branch from this branch so that you can look at it before merging back into this branch?

Reply:
That works for me, thanks.

Reply:
Resolved by #526